contracts/SimpleYieldEulerLever.sol

```diff
@@ -0,0 +1,395 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.14;
+
+import "@yield-protocol/utils-v2/contracts/cast/CastU128I128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastI128U128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I256.sol";
+import "@yield-protocol/utils-v2/contracts/token/TransferHelper.sol";
+import "./interfaces/IEulerMarkets.sol";
+import "./interfaces/IEulerEToken.sol";
+import "./YieldLeverBase.sol";
+
+/// @title A simple euler lever designed to work for one euler token & its underlying at a time
+/// @author iamsahu
+/// @notice Working:
+///         - Get flash loan of USDC/DAI/WETH
+///         - Deposit to get eulerToken
+///         - Deposit & borrow against it
+///         - Sell the fyToken to get USDC/DAI/WETH
+///         - Close the flash loan
+contract SimpleYieldEulerLever is YieldLeverBase {
+    using TransferHelper for IERC20;
+    using CastU128I128 for uint128;
+    using CastI128U128 for int128;
+    using CastU256U128 for uint256;
+    using CastU256I256 for uint256;
+
+    /// @notice euler market
+    IEulerMarkets public constant markets =
+        IEulerMarkets(0x3520d5a913427E6F0D6A83E07ccD4A4da316e4d3);
+    /// @notice Euler protocol address
+    address constant EULER = 0x27182842E098f60e3D576794A5bFFb0777E025d3;
+    /// @notice Asset underlying the Euler Token
+    address immutable asset;
+    /// @notice Join of the underlying asset
+    FlashJoin immutable assetJoin;
+    /// @notice Euler token for the asset
+    IEulerEToken immutable eToken;
+    /// @notice ilkId of the asset
+    bytes6 immutable ILKID;
+    /// @notice Join of the ilk
+    address immutable ilkJoin;
+
+    constructor(
+        bytes6 ilkId_,
+        Giver giver_,
+        address asset_,
+        FlashJoin assetJoin_
+    ) YieldLeverBase(giver_) {
+        // Approve the main euler contract to pull your tokens:
+        IERC20(asset_).approve(EULER, type(uint256).max);
+        // Approve join for
+        IERC20(asset_).approve(address(assetJoin_), type(uint256).max);
+        assetJoin = assetJoin_;
+        asset = asset_;
+        ILKID = ilkId_;
+
+        eToken = IEulerEToken(markets.underlyingToEToken(asset_));
+        ilkJoin = address(ladle.joins(ILKID));
+    }
+
+    /// @notice Approve maximally for an fyToken.
+    /// @param seriesId The id of the pool to approve to.
+    function approveFyToken(bytes6 seriesId) external {
+        IPool(ladle.pools(seriesId)).fyToken().approve(
+            address(ladle),
+            type(uint256).max
+        );
+    }
+
+    /// @notice Invest by creating a levered vault.
+    /// @param seriesId The series to create the vault for.
+    /// @param baseAmount The amount of own liquidity to supply.
```

Comment:
```suggestion
    /// @param ink The amount of own eToken to supply as collateral.
```

Please replace `baseAmount` elsewhere, as we would use that name for a borrowable asset, not a collateral.